### ISSUE TITLE
Lazy timers

### ISSUE THREAD

**Issue by aaron02** [AscEmu/AscEmu#433]
Test script:

Isexpired wont work ?

```
LazyTimer Timer;

MOONSCRIPT_FACTORY_FUNCTION(TestAI, MoonScriptAI);
TestAI(Creature* pCreature) : MoonScriptBossAI(pCreature), Timer(5000)
{}

Void OnCombatStart()
{
    Timer.start();
}

Void AIUpdate()
{
    Timer.evaluate();
    If(Timer.isExpired())
    {
        Emote("test", Text_Yell, 0);
        Timer.reset(5000);
    }

    ParentClass::AIUpdate();
}
```

By adding 
```
If(Timer.isExpired() == 2 )

```
Wont work

**Comment by Zyres** [comment]
Timer.isExpired() returns a bool (true (1) or false (0))
If you comparing it with 2 the result will be always false.

**Comment by aaron02** [comment]
When doing it as in the script above the test text spams after finishing the 5 seconds. The timerstate changes to stop.

**Comment by Evairfairy** [comment]
You're using the wrong API, `reset` is just to reset the timer back to it's initial state; you need to also call `start` to restart it.

That said, it seems like a common task, so I've added the `resetAndRestart` API call - use it exactly how you're using `reset` in the above sample.

Additionally, there was a timing bug that would cause the duration between updates to increase exponentially; this has now been fixed in aa61273bb608f8ab6eaaf8a01c131379e19428bb
